source/MaterialXGenShader/GenContext.h

```diff
@@ -223,6 +241,12 @@ class GenContext
 
     // List of output suffixes
     std::unordered_map<const ShaderOutput*, string> _outputSuffix;
+
+    // Texture baking data
+    bool _bake = false;
```

Comment:
It turns out that I actually need to leverage the genContext class because when I call the shader generator to create the shader it uses the genContext class (not the genOptions) already when creating the shader. Should I add genOptions when the shader is generated?

Reply:
We discussed this briefly in person, but I believe the GenOptions instance is a data member of GenContext, which would give you access at shader generation time.